test/e2e/simple_test.go

```diff
@@ -120,3 +122,90 @@ func TestFailingBuild(t *testing.T) {
 		t.Fatalf("watchBuild did not return expected error: %v", err)
 	}
 }
+
+func TestBuildLowTimeout(t *testing.T) {
+	clients := setup(t)
+
+	buildName := "build-low-timeout"
+	buildTimeout := "50s"
+	if _, err := clients.buildClient.builds.Create(&v1alpha1.Build{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: buildTestNamespace,
+			Name:      buildName,
+		},
+		Spec: v1alpha1.BuildSpec{
+			Timeout: buildTimeout,
+			Steps: []corev1.Container{{
+				Name:    "lowtimeoutstep",
+				Image:   "ubuntu",
+				Command: []string{"/bin/bash"},
+				Args:    []string{"-c", "sleep 2000"},
+			}},
+		},
+	}); err != nil {
+		t.Fatalf("Error creating build: %v", err)
+	}
+
+	b, err := clients.buildClient.watchBuild(buildName)
+	if err == nil {
+		t.Error("watchBuild did not return expected BuildTimeout error")
+	}
+
+	if &b.Status == nil {
+		t.Fatalf("wanted build status to be set; got nil")
+	}
+
+	successCondition := b.Status.GetCondition(v1alpha1.BuildSucceeded)
+
+	if successCondition == nil {
+		t.Fatalf("wanted build status to be set; got %q", b.Status)
+	}
+
+	// verify reason for build failure is timeout
+	if successCondition.Reason != "BuildTimeout" {
+		t.Fatalf("wanted BuildTimeout; got %q", successCondition.Reason)
+	}
+	buildDuration := b.Status.CompletionTime.Time.Sub(b.Status.StartTime.Time).Seconds()
+	lowerEnd, err := time.ParseDuration(buildTimeout)
+	if err != nil {
+		t.Errorf("Error parsing build duration: %v", err)
+	}
+	higherEnd := 90 * time.Second // build timeout + 30 sec poll time + 10 sec
```

Comment:
If this value is intended to be based on `buildTimeout`, maybe it should explicitly be based on it?

`higherEnd := buildTimeout + 30*time.Second + 10*time.Second`

That way if we change `buildTimeout` above we won't also have to remember to update this value.